src/components/interaction-designer/resource-editors/ResourceEditor.vue

```diff
@@ -225,6 +231,16 @@ export class ResourceEditor extends mixins(FlowUploader, Permissions, Routes, La
     }
   }
 
+  /**
+   * Compute resource index (cell index) for a table having X languages and Y modes
+   *
+   * @param langIndex
+   * @param modeIndex
+   */
+  computeResourceIndex(langIndex: number, modeIndex: number): number {
+    return langIndex * this.flow.supported_modes.length + modeIndex
+  }
```

Comment:
As we discussed, `computeResourceIndex` is currently fragile (if we change the number of modes for the flow).
It would be great to find a way to make it more robust.

Reply:
@bulatgab , thanks for raising flag here. I do agree using number `index` here could be confusing, but that's how the jsonSchema validate values of array (`resources.values` is an array).
I spent time to find alternative options, but couldn't so far.
Fortunately, I was able to correct the resources.values when users update `modes`; but it gets a bit tricky when they update languages. 

Therefore, I created a new ticket to handle that: https://viamoinc.atlassian.net/browse/VMO-5213

Reply:
I also have some concerns here. It looks like you are making the index of resource language modes predictable by sorting when languages and modes change and then using the predictability to look up using an integer index. This is going to be fragile whatever I think - what if someone forgets to sort in future in a new place these values are changed? It also doesn't seem to work as intended though as the resource languages order doesn't seem to match this sorted order?

I understand this may be fixed in 5213. Something we might want to consider then is running a separate validation for values outside of the full resource validation. That way we can validate more specifically based on the content type (e.g. only allow a file path for certain content types) and we can also key the validation errors however we want rather than relying on the structure of the resource. For example, we can have string keys based on the resource uuid/mode/language id combination rather than relying on the order of an array. Does this make sense? Maybe this would be worse as it would involve a pretty different approach to what we use elsewhere but maybe also worth considering to remove the sort and resolve other issues here?

Reply:
@jcbashdown , thanks for input here.
My main blocker is that if we validate an array using AJV, the returned corresponding `dataPath` is `numeric indexed`, see this vuex state for eg:
![Screen Shot 2022-01-25 at 3 49 22 PM](https://user-images.githubusercontent.com/68745918/151065832-f0bb9e25-5758-40fd-8576-dc8ebdf1a9d9.png)

The numeric index corresponds to the order of the Data in the array, that's why it's important to follow a certain order. I know this is quite fragile, and I'm open if we can find alternative. The only solution I found so far:
- force order (like I did in this PR)
- change the resources spec from `array` to object

Any suggestion?

Reply:
The issue I was finding - the actual ordering of the language values in the resource on your branch are not consistent. They still depend on the order in which I add the language. That means I can add the second language in the list on creation and then go back and add the first. Now when I edit the second tab in the UI the first becomes valid and vice versa.

This issue isn't quite the same as the one in VMO-5213 as that's about the fact that existing resources don't get new empty values when you add a new language.

The solution I'm proposing for the issue I've found here is that we are probably going to need to validate resource value strings directly and build up our own nested structure of validation errors corresponding to the IDs of language and modes. This is much more robust to edits of languages and modes than what we have but sacrifices the convenience of using the json schema for resources. It has the advantage though, of potentially letting us create specific validators for each content type.

Let's try and have a call at some point to discuss further.

Reply:
As discussed in our call, we will handle languages related behaviour in https://viamoinc.atlassian.net/browse/VMO-5213